ONSdigital/rm-actionexporter-service
# upgrade to spring boot2

# Motivation and Context

Spring Boot 1.x had its end-of-life in August 2019.
Maintenance for Spring Boot 2.1 will end 1 November 2020, so it's time to upgrade Spring Boot to 2.2.
Spring Boot 2 also brings some useful changes, including a welcome simplification of security settings.

# What has changed
- Updated Spring Framework to 5.2, and Spring Boot to 2.2.
- Updated lib dependencies, migrated deprecated config in YAML, with security config in YAML replaced by a new class SecurityConfig.java.
- copied in missing `ctp.common.error.CTPRetryPolicy` from Action Service, and updated its imports for SB 2.x
- replaced godaddy-logger with slf4j




# How to test?
- `mvn clean install`
- `mvn spring-boot:run`
- deploy gateway with new image to a sandbox cluster
- set bucket in sandbox to a test bucket using ras-rm-cli
- run acceptance tests on sandbox cluster 
- use port forwarding to connect to Response Ops on localhost:8085, and create a new collection exercise for same day
- check that the print file is in the test bucket 




# Links
- https://collaborate2.ons.gov.uk/jira/browse/RAS-82
